controllers/machinepool_controller.go

```diff
@@ -57,13 +76,181 @@ func (r *MachinePoolReconciler) SetupWithManager(mgr ctrl.Manager, options contr
 	r.controller = c
 	r.recorder = mgr.GetEventRecorderFor("machinepool-controller")
 	r.config = mgr.GetConfig()
-
 	r.scheme = mgr.GetScheme()
 	return nil
```

Comment:
Pending resolution of https://github.com/kubernetes-sigs/cluster-api/pull/2101/files#r371995294, may need to add a watch map function from Cluster -> MachinePool

Reply:
We should likely also port over the immutability check on `ClusterName` as well.

Reply:
The immutability check is part of a validation webhook @detiber? Would it be okay to enable the validation webhooks and immutability check in a follow-up PR?

Reply:
I'm not sure I fully understand what's required here. @vincepri @detiber could either of you clarify?

Reply:
I'm not totally sure (the link to the discussion is not working), but do we need to receive reconcile requests here when an Cluster object is updated?

Reply:
I have not encountered the need for triggering when cluster is updated during my testing, but I might not be exercising the scenario ya'll have in mind. @vincepri @detiber @ncdc Could I get some clarification or can we address this in a subsequent PR?